src/meshes/transfinite_mappings_3d.jl

```diff
@@ -0,0 +1,172 @@
+#      │             │                                      │             │
+#      │             │                                      │             │
+#      │    6 (-x)   │                                      │             │
+#      │             │                                      │             │
+#      │             │                                      │             │
+#      │             │                                      │             │
+#      │             │                                      │             │
+#      │             ④────────────────────────{5}──────────│─────────────③
+#      │            ╱                                       │            ╱
+#      │           ╱              1 (-y)                    │           ╱
+#     {4}         ╱                                        {2}         ╱
+#      │         ╱                                          │         ╱
+#      │        ╱                                           │        ╱
+#      │      {9}                                           │      {10}
+#      │      ╱                                             │      ╱
+#      │     ╱                                              │     ╱  Global coordinates:
+#      │    ╱                                               │    ╱       z
+#      │   ╱                             3 (-z)             │   ╱        ↑   y
+#      │  ╱                                                 │  ╱         │  ╱
+#      │ ╱                                                  │ ╱          │ ╱
+#      │╱                                                   │╱           └─────> x
+#      ①───────────────────────{1}─────────────────────────②
+
+
+# Transfinite mapping formula from a point (xi, eta, zeta) in reference space [-1,1]^3 to a
+# physical coordinate (x, y, z) for a hexahedral element with straight sides
+function straight_side_hex_map(xi, eta, zeta, corner_points)
+
+  coordinate = zeros(eltype(xi), 3)
+  for j in 1:3
+    coordinate[j] += (0.125 * ( corner_points[j, 1] * (1.0 - xi) * (1.0 - eta) * (1.0 - zeta)
+                              + corner_points[j, 2] * (1.0 + xi) * (1.0 - eta) * (1.0 - zeta)
+                              + corner_points[j, 3] * (1.0 + xi) * (1.0 + eta) * (1.0 - zeta)
+                              + corner_points[j, 4] * (1.0 - xi) * (1.0 + eta) * (1.0 - zeta)
+                              + corner_points[j, 5] * (1.0 - xi) * (1.0 - eta) * (1.0 + zeta)
+                              + corner_points[j, 6] * (1.0 + xi) * (1.0 - eta) * (1.0 + zeta)
+                              + corner_points[j, 7] * (1.0 + xi) * (1.0 + eta) * (1.0 + zeta)
+                              + corner_points[j, 8] * (1.0 - xi) * (1.0 + eta) * (1.0 + zeta) ) )
+  end
+
+  return coordinate
+end
+
+
+# Construct the (x, y, z) node coordinates in the volume of a straight sided hexahedral element
+function calc_node_coordinates!(node_coordinates::AbstractArray{<:Any, 5}, element, nodes, corners)
```

Comment:
Have you checked that `corners` is used consistently, e.g., that it is not called `vertices` in the 2D part etc.?

Reply:
No yet. I need to look through the code and check the naming conventions. In the docs I use corners consistently (in the figures and discussions).

Reply:
I unified the notation of this using a compromise. In the transfinite interpolation routines I use the name `corner` to match the mesh docs whereas in the `p4est_mesh.jl` all my new routines use the term `vertex` or `vertices`.

Reply:
Hm. Not a big fan of such inconsistencies, but maybe a hiwi can resolve this in the future. Thus, feel free to resolve this thread (or create an issue for it).

Reply:
If we want to abolish the word `corner` this would require updating most of the 2D/3D unstructured documentation as well as comments in said code. The main issue here (I think) is that the `p4est` library uses vertex/vertices whereas nearly every else we refer to these values as corners. That being said, the `p4est` structs also have corners but they mean something different than what we typically mean (I think). So I agree this is somewhat confusing but trying to unify notation and variable naming conventions between Trixi and `p4est` seems like tilting at windmills to me.

Reply:
OK, you have me convinced that consistency is either hard to achieve or not desirable/achievable at all. I'll resolve this conversation here, but discussing it next week might be a good idea.